src/CacheWithResilience.php

```diff
@@ -0,0 +1,165 @@
+<?php
+declare(strict_types=1);
+
+namespace BBC\ProgrammesCachingLibrary;
+
+use DateTime;
+use DateInterval;
+use DateTimeInterface;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Cache\CacheItem;
+
+/**
+ * An implementation of stale-if-error for the Cache library.
+ * This class is compatible with BBC\ProgrammesCachingLibrary\Cache, the only difference when calling the functions
+ * is that you can call getItem() with an extra parameter to get the stale value.
+ */
+class CacheWithResilience extends Cache
+{
+    /** @var LoggerInterface  */
+    private $logger;
+
+    /** @var \Exception[]  */
+    private $whitelistedExceptions;
+
+    // How long items will persist in the cache server.
+    // This also mean for how long we could serve stale content in case of issues
+    private const RESILIENCE_TTL = 21600; // 6h
```

Comment:
I think this should probably be passed in via the DI rather than be a class constant, we may want to change it between projects.